lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1235,6 +1235,29 @@ func (l *Log) convertFieldsToDynamoMapFormat(ctx context.Context) error {
 	return trace.Wrap(l.migrateMatchingEvents(ctx, filterExpr, transformEvent))
 }
 
+func minInt64(x, y int64) int64 {
+	if x > y {
+		return y
+	}
+	return x
+}
+
+func (l *Log) approximateOptimalMigrationWorkers() (int64, error) {
+	req := dynamodb.DescribeTableInput{TableName: aws.String(l.Tablename)}
+	table, err := l.svc.DescribeTable(&req)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	// calculate the throughput, accounting for r/w bottlenecks
+	provisioned := table.Table.ProvisionedThroughput
+	throughput := minInt64(*provisioned.ReadCapacityUnits, *provisioned.WriteCapacityUnits)
```

Comment:
No chance any either of these pointers could be nil?

Reply:
`provisioned` may actually be nil, added a check that defaults `MaxMigrationWorkers`. This happens when the DB is in pay-on-demand mode.

Reply:
I don't know if this is possible but could either `ReadCapacityUnits` or `WriteCapacityUnits` be nil, if only one of them is provisioned or something? Just don't want to potentially panic here based on something returned from AWS which is basically external input

Reply:
So the SDK doesn't explicitly say whether these fields are optional or not. I assumed they're not and are always there but I extended the check above just to make sure.